src/os_vxworks.c

```diff
@@ -1,146 +1,201 @@
+#endif /* if(defined(_WRS_KERNEL) */
 
-os_lib_handle os_library_open(
-	const char *path )
+/* directory operations */
+os_uint64_t os_directory_free_space( const char* path )
 {
 	return 0;
 }
 
-os_status_t os_library_close(
-	os_lib_handle lib )
+/** @todo fix later once it needs in vxWorks */
+os_status_t os_directory_delete(
+	const char *path, const char *regex, os_bool_t recursive )
 {
-	return OS_STATUS_FAILURE;
-}
+	os_status_t result = OS_STATUS_FAILURE;
 
-void *os_library_find(
-	os_lib_handle lib,
-	const char *function )
-{
-	return NULL;
-}
+	if (regex == NULL && rmdir (path) == OK)
+		result = OS_STATUS_SUCCESS;
 
-os_uint64_t os_directory_free_space( const char* path )
-{
-	return 0;
+	return result;
 }
 
-os_status_t os_file_temp(
-	char *prototype,
-	size_t suffix_len)
+/* file operations */
+os_status_t os_file_chown(
+	const char *path,
+	const char *user )
 {
 	return OS_STATUS_FAILURE;
 }
 
-os_status_t os_process_cleanup( void )
+
+/* process functions */
+os_status_t os_path_executable(
+	char *path,
+	size_t size )
 {
-	return 0;
+	os_status_t result = OS_STATUS_BAD_PARAMETER;
+	if ( path )
+	{
+		const char *task_name = taskName(taskIdSelf());
+		result = OS_STATUS_FAILURE;
+		if (task_name && size > 2u)
+		{
+			path[0] = OS_DIR_SEP;
+			strncpy(&path[1], task_name, size - 1u);
+			result = OS_STATUS_SUCCESS;
+		}
+	}
+	return result;
 }
 
-os_status_t os_file_chown(
-	const char *path,
-	const char *user )
+os_status_t os_process_cleanup( void )
 {
-	return 0;
+	return OS_STATUS_FAILURE;
 }
 
-/** @todo fix later once it needs in vxWorks */
-os_status_t os_directory_delete(
-	const char *path, const char *regex, os_bool_t recursive )
+/* service functions */
+os_status_t os_service_run(
+	const char *id,
+	os_service_main_t service_function,
+	int argc,
+	char *argv[],
+	int remove_argc,
+	const char *remove_argv[],
+	os_sighandler_t UNUSED(handler),
+	const char *UNUSED(logdir) )
 {
-	return 0;
+	os_status_t result = OS_STATUS_BAD_PARAMETER;
+	if ( id && service_function )
+	{
+		int i;
+		char** good_argv = (char**)malloc( (unsigned long)argc * sizeof( char* ) );
+		result = OS_STATUS_FAILURE;
+		/* remove bad arguments */
+		if ( good_argv )
+		{
+			int good_argc = 0u;
+			for ( i = 0; i < argc; ++i )
+			{
+				int is_good_arg = 1; /* true */
+				if ( remove_argc > 0 && remove_argv && argv[i] )
+				{
+					int j;
+					for ( j = 0;
+						j < remove_argc && is_good_arg; ++j )
+					{
+						if ( remove_argv[j] )
+						{
+							size_t arg_len =
+								strlen( remove_argv[j] );
+							is_good_arg = strncmp( argv[i],
+								remove_argv[j], arg_len );
+						}
+					}
+				}
+
+				if ( is_good_arg )
+				{
+					good_argv[good_argc] = argv[i];
+					++good_argc;
+				}
+			}
+
+			if ( ( *service_function )( good_argc, good_argv ) == EXIT_SUCCESS )
+				result = OS_STATUS_SUCCESS;
+
+			free( good_argv );
+		}
+	}
+	return result;
 }
 
 /** @todo fix later once it needs in vxWorks */
```

Comment:
might as well fix the grammar in the comment

Reply:
fixed